packages/transmute-did/src/lib/openpgpExtensions/did/ocap/json-ld/__tests__/webOfTrustDemo.spec.js

```diff
@@ -0,0 +1,249 @@
+
+    AtoBCapWithCavs = await openpgpSignJson(objCap, ADoc.publicKey[0].id, A.privateKey, passphrase);
+    // console.log(AtoBCapWithCavs);
+  });
+
+  it('can B delegate his partial access to C to D with caveats?', async () => {
+    const objCap = {
+      '@context': [
+        'https://example.org/did/v1',
+        'https://example.org/ocap/v1',
+        'http://schema.org',
+      ],
+      id: 'did:example:d2c83c43-878a-4c01-984f-b2f57932ce5f',
+      type: 'Proclamation',
+
+      // Yet again, point up the chain...
+      parent: AtoBCapWithCavs.id,
+
+      // Now we grant access to one of Dummy Bot's keys
+      grantedKey: DDoc.publicKey[0].id,
+
+      // We add a new caveat/attenuation: this one will expire 30 days
+      // in the future
+      caveat: [
+        {
+          id: 'did:example:d2c83c43-878a-4c01-984f-b2f57932ce5f#caveats/expire-time',
+          type: 'ExpireTime',
+          date: '2017-09-23T20:21:34Z',
+        },
+      ],
+
+      // Finally we sign this object with Bob's key
+      //   This will be done by bob below
+      //   signature: {
+      //     type: "RsaSignature2016",
+      //     created: "2016-02-08T17:12:28Z",
+      //     creator: "did:example:ee568de7-2970-4925-ad09-c685ab367b66#key-1",
+      //     signatureValue: "..."
+      //   }
+    };
+    BtoDCapWithCavs = await openpgpSignJson(objCap, BDoc.publicKey[0].id, B.privateKey, passphrase);
+    // console.log(BtoDCapWithCavs);
+  });
+
+  it('can D use cap chain to upload to C?', async () => {
+    const objCap = {
+      '@context': [
+        'https://example.org/did/v1',
+        'https://example.org/ocap/v1',
+        'http://schema.org',
+      ],
+      id: 'did:example:2bdf6273-a52e-4cdf-991f-b5f000008829',
+      type: 'Invocation',
+
+      // Dummy Bot is invoking the proclamation it has,
+      // but the whole chain will be checked for attenuation and
+      // verification of access
+      proclamation: BtoDCapWithCavs.id,
+
+      // The method being used
+      method: 'UploadFile',
+
+      // The key Dummy Bot is using in this invocation
+      usingKey: DDoc.publicKey[0].id,
+
+      // Here's the base64 encoded file as part of the payload
+      file: base64url('some plaintext from D'),
+
+      // Finally we sign this object with Dummy Bot's key
+      //   Signed by D below
+      //   signature: {
+      //     type: "RsaSignature2016",
+      //     created: "2016-02-08T17:13:48Z",
+      //     creator: "did:example:5e0fe086-3dd7-4b9b-a25f-023a567951a4#key-1",
+      //     signatureValue: "..."
+      //   }
+    };
+
+    DInvocOfC = await openpgpSignJson(objCap, DDoc.publicKey[0].id, D.privateKey, passphrase);
+    // console.log(DInvocOfC);
+  });
+
+  it('can D send encrypted Invocation to C?', async () => {
+    secureMessageForC = await cryptoHelpers.encryptMessage({
+      message: JSON.stringify(DInvocOfC),
+      privateKey: D.privateKey,
+      publicKey: C.publicKey,
+      passphrase,
+    });
```

Comment:
Should there be an assertion for Truthy?

Reply:
fixed